lib/controllers/frontend/spree/users_controller.rb

```diff
@@ -25,12 +25,12 @@ def create
   end
 
   def update
-    if @user.update_attributes(user_params)
-      if params[:user][:password].present?
-        # this logic needed b/c devise wants to log us out after password changes
-        user = Spree::User.reset_password_by_token(params[:user])
-        sign_in(@user, :event => :authentication, :bypass => !Spree::Auth::Config[:signout_after_password_change])
-      end
+    successful_update = @user.has_password? ? @user.update_with_password(user_with_password_params) : @user.update_without_password(user_params) && @user.set_has_password!
```

Comment:
Can we please move this slightly complex logic to it's own private method? This will make it more readable and concise. You could even name the method `successful_update` so that the if statement stands as-is and the variable can just go away.

If you do move to the method, I would actually name it `successful_update?`